### PR TITLE
Reduce nested API calls

### PR DESCRIPTION
**Pull request by debecerra**
- reduce the number of API calls being performed by getting author data directly from the database if that author data exists on our server
- currently, this will occur for just LocalAuthors (which is enough to keep Heroku from crashing from too many nested calls to our API)
- later we can also cache author data from remote authors and get that author data if it is recent
- also added one additional fix of increasing the max length of a few CharFiield fields which I think is causing a crash on Heroku

### CODE REVIEW

**Review by olivier-vadiaval — APPROVED** [review]

**Review by JawadHossain — APPROVED** [review]
Awesome!

File: code/api/tests/test_followers.py [hunk 1]
```diff
@@ -109,13 +108,29 @@ def setUpClass(cls):
     def tearDownClass(cls):
         logging.disable(logging.NOTSET)
 
-    def test_get_followers(self):
+    def test_get_followers_empty(self):
```
Comment (JawadHossain) [inline code/api/tests/test_followers.py:111]: wait, i think this test and next test have same names. can you pls check

**Review by JawadHossain — CHANGES_REQUESTED** [review]
one comment

File: code/api/tests/test_followers.py (hunk 1 above)
Comment (debecerra) [reply code/api/tests/test_followers.py:111]: Good catch! fixed it now

**Review by JawadHossain — APPROVED** [review]

**Pull request merged**